### PR TITLE
Fix dependency check

### PR DESCRIPTION
**Pull request by sharkwouter**
The dependency check and the README were missing some dependencies which are needed. I found this while trying to make a prepare script for Fedora.

The which check may look a bit odd, but the Fedora docker container doesn't have it.

This includes the changes of #124.

### CODE REVIEW

**Comment by sharkwouter** [comment]
Just to clarify, this does enforce some dependencies for psp-pacman as well, like libssl, libarchive, libgpgme and libcurl.

**Comment by sharkwouter** [comment]
I've now removed the tcl check. This assumes that the requirement for tcl will changed in the near future.

**Comment by sharkwouter** [comment]
Okay, I've reverted to tcl being listed for now.

File: depends/check-dependencies.sh [hunk 1]
```diff
@@ -57,30 +57,40 @@ check_header    libelf          elf.h libelf.h libelf/libelf.h gelf.h libelf/gel
...
+check_header    libcurl         curl/curl.h
+check_header    libssl          openssl/ssl.h
 
 check_program   git
 check_program   svn
 check_program   wget
 check_program   patch
 check_program   tar
 check_program   unzip
+check_program   bzip2
```
Comment (carstene1ns) [inline depends/check-dependencies.sh:70]: Might want to add xz here.

File: depends/check-dependencies.sh [hunk 2]
```diff
@@ -57,30 +57,40 @@ check_header    libelf          elf.h libelf.h libelf/libelf.h gelf.h libelf/gel
...
 check_program   gcc
 check_program   g++
-
-# check_program   freetype-config
+check_program   m4
 
 check_program   bison
 check_program   flex
 check_program   tclsh
+check_program   cmp
```
Comment (carstene1ns) [inline depends/check-dependencies.sh:83]: Do you happen to know where cmp is needed?

File: depends/check-dependencies.sh [hunk 3]
```diff
@@ -57,30 +57,40 @@ check_header    libelf          elf.h libelf.h libelf/libelf.h gelf.h libelf/gel
...
+check_program   which
 
 check_program   makeinfo
 check_program   doxygen
 
+check_program   gpgme-config
```
Comment (carstene1ns) [inline depends/check-dependencies.sh:89]: Might be better to look for `gpgme.h`.

File: depends/check-dependencies.sh (hunk 2 above)
Comment (sharkwouter) [reply depends/check-dependencies.sh:83]: I think it's used when building newlib, but I'm not 100% sure. I can test again later.

File: depends/check-dependencies.sh (hunk 1 above)
Comment (sharkwouter) [reply depends/check-dependencies.sh:70]: Good point, will do

File: depends/check-dependencies.sh (hunk 3 above)
Comment (sharkwouter) [reply depends/check-dependencies.sh:89]: True, will change it

File: depends/check-dependencies.sh (hunk 2 above)
Comment (sharkwouter) [reply depends/check-dependencies.sh:85]: GCC actually doesn't build without diffutils: https://gist.github.com/sharkwouter/5a0a29e90de541f3597f67677c1fd43d

I'll change it to diff, though, more well known and part of the same toolset.

**Comment by carstene1ns** [comment]
LGTM now.
However, will investigate the tcl dependency, as we explicitly disable it:
https://github.com/pspdev/psplibraries/blob/e7126df5ffda9e304e2a0b27cdd60a6c4b150ebc/scripts/sqlite.sh#L6

**Comment by sharkwouter** [comment]
Cool, keep me posted

**Pull request merged**